core/src/test/scala/doric/matchers/CustomMatchers.scala

```diff
@@ -0,0 +1,116 @@
+package doric.matchers
+
+import doric.sem.{DoricMultiError, DoricSingleError}
+import org.apache.spark.sql.AnalysisException
+import org.scalatest.matchers._
+
+/**
+  * Object to ease the import
+  */
+object CustomMatchers extends CustomMatchers
+
+/**
+  * Custom Doric errors matcher
+  */
+trait CustomMatchers {
+  private type DSE = DoricSingleError
+
+  private def msgErrors(
+      dme: Set[DSE],
+      expectedDme: Set[DSE]
+  ): Either[String, Option[String]] = {
+
+    if (dme.size == expectedDme.size) {
+      val errors: Set[Either[(DSE, DSE), DSE]] = dme
+        .zip(expectedDme)
```

Comment:
We can't zip two sets, since there is no guarantee that the elements are sorted. 

Instead, I'd return a tuple with three components: common errors, actual unexpected errors, and missing expected errors.